extension/src/content_scripts/contentView.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * GPII Chrome Extension for Google Chrome
+ *
+ * Copyright 2017-2018 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this license.
+ *
+ * You may obtain a copy of the license at
+ * https://github.com/GPII/gpii-chrome-extension/blob/master/LICENSE.txt
+ */
+
+/* global fluid, gpii */
+"use strict";
+
+(function ($, fluid) {
+
+    /*
+     * `gpii.chrome.contentView` is a type of view component which extends the typical viewComponent faculties for
+     * searching for selectors. The extended features allow for searching for "content" amongst a variety of selectors,
+     * and returning the first one found. This is useful when trying to determine the primary content of a page, and is
+     * typically used by enactors acting upon an entire document without explicit knowledge of the primary content area.
+     *
+     * The content is located upon component creation and stored in the `content` member. The `content` option can be
+     * set with an array of selector names to determine which selectors should be used to search for the content. The
+     * order determines which selectors to look through first. The search will terminate after the first selector
+     * that returns elements. If none of the selectors is able to locate any elements, the value stored in the
+     * `defaultContent` is returned or if that is not provided, an empty jQuery object is returned.
+     *
+     * Additionally, methods are provided to find selectors scoped within the content (`locateInContent`) or scoped to
+     * the container, but outside of the content (`locateOutOfContent`). These methods behave similarly to calls to
+     * `locate`; which is also still available for use. However, they cannot be used in IoC expressions like `locate`
+     * can.
+     */
+    fluid.defaults("gpii.chrome.contentView", {
+        gradeNames: ["fluid.viewComponent"],
+        selectors: {
+            article: "article, [role~='article'], .article, #article",
+            main: "main, [role~='main'], .main, #main",
+            genericContent: ".content, #content, .body:not('body'), #body:not('body')"
+        },
+        content: ["article", "main", "genericContent"],
+        defaultContent: "",
+        members: {
+            content: {
+                expander: {
+                    funcName: "gpii.chrome.contentView.findFirstSelector",
+                    args: ["{that}", "{that}.options.content", "{that}.options.defaultContent"]
+                }
+            }
+        },
+        invokers: {
+            locateInContent: {
+                funcName: "gpii.chrome.contentView.locateInContent",
+                args: ["{that}", "{arguments}.0"]
+            },
+            locateOutOfContent: {
+                funcName: "gpii.chrome.contentView.locateOutOfContent",
+                // the last argument isn't directly used, but is provided to force IoC resolution of the method which is
```

Comment:
How is it that this function ends up being called before component creation is done?